### PR TITLE
[PLAYER-1724] Advanced Player Controls

### PR DESCRIPTION
**Pull request by AlinaVoronkovaEpam**
[PLAYER-3712] Playback-rate: Communication between skin and sdk.
[PLAYER-4298] Playback-speed range dependencies were removed from sdk

Please merge sdk and skin branches simultaneously.

### CODE REVIEW

**Comment by pilievOoyala** [comment]
LGTM

File: sdk/android/skin/src/main/java/com/ooyala/android/skin/OoyalaSkinBridgeEventHandlerImpl.java [hunk 1]
```diff
@@ -160,6 +160,13 @@ public void onAudioTrackSelected(ReadableMap parameters) {
     _player.setUserDefinedAudioTrack(parameters.getString("audioTrack"));
   }
 
+  @Override
+  public void onPlaybackSpeedRateSelected(ReadableMap parameters) {
+    Dynamic dynamic = parameters.getDynamic("playbackSpeedRate");
```
Comment (SZadorozhnyiEpam) [inline sdk/android/skin/src/main/java/com/ooyala/android/skin/OoyalaSkinBridgeEventHandlerImpl.java]: Can it be extracted as a static string?

Comment (SZadorozhnyiEpam) [inline sdk/android/skin/src/main/java/com/ooyala/android/skin/OoyalaSkinBridgeEventHandlerImpl.java]: Can it be extracted as static field?

**Review by SZadorozhnyiEpam — CHANGES_REQUESTED** [review]
Left a comment

**Comment by akshaydch11** [comment]
As this feature is targeted for 4.40 and 4.39 is small release so we don't want to take chances of introducing new bugs by merging this into candidate. So, lets hold off merging till we release 4.39.

File: sdk/android/skin/src/main/java/com/ooyala/android/skin/OoyalaSkinBridgeEventHandlerImpl.java (hunk 1 above)
Comment (AlinaVoronkovaEpam) [reply sdk/android/skin/src/main/java/com/ooyala/android/skin/OoyalaSkinBridgeEventHandlerImpl.java]: done

**Review by SZadorozhnyiEpam — APPROVED** [review]

File: sdk/android/skin/src/main/java/com/ooyala/android/skin/OoyalaSkinBridgeEventHandlerImpl.java [hunk 2]
```diff
@@ -160,6 +161,13 @@ public void onAudioTrackSelected(ReadableMap parameters) {
     _player.setUserDefinedAudioTrack(parameters.getString("audioTrack"));
   }
 
+  @Override
+  public void onPlaybackSpeedRateSelected(ReadableMap parameters) {
+    Dynamic dynamic = parameters.getDynamic(PLAYBACK_SPEED_RATE);
+    String speed = dynamic.asString();
+    _player.setSelectedPlaybackSpeed(Float.parseFloat(speed));
```
Comment (brol1dev) [inline sdk/android/skin/src/main/java/com/ooyala/android/skin/OoyalaSkinBridgeEventHandlerImpl.java:168]: use `OoyalaPlayer.setPlaybackSpeed(float)` instead now and we can remove `setSelectedPlaybackSpeed`

**Review by brol1dev — COMMENTED** [review]
I added one comment. I think I'm going to merge the PR to create a RC but please fix and also what I commented in android-sdk repo.

**Review by brol1dev — APPROVED** [review]
LGTM

**Pull request merged**